src/lib/statistics/abstract_statistics_object.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <utility>
+
+#include "all_type_variant.hpp"
+#include "cardinality.hpp"
+#include "cardinality_estimate.hpp"
+#include "selectivity.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * Base class for types that hold statistical information about a column/segment of data.
+ */
+class AbstractStatisticsObject {
+ public:
+  explicit AbstractStatisticsObject(const DataType data_type);
+  virtual ~AbstractStatisticsObject() = default;
+
+  /**
+   * @return A statistics object that represents the data after the predicate has been applied.
+   */
+  virtual std::shared_ptr<AbstractStatisticsObject> sliced(
```

Comment:
because the object itself is not manipulated, which "slice" implies, but instead the function returns a "sliced"-off portion of the statistics object.